course/src/main/java/in/testpress/course/viewmodels/OfflineAttachmentViewModel.kt

```
@@ -81,3 +86,58 @@ class OfflineAttachmentViewModel(application: Application) : AndroidViewModel(ap
+            file
+        )
+    }
+
+    val intent = Intent(Intent.ACTION_VIEW).apply {
+        setDataAndType(uri, context.contentResolver.getType(uri))
+        flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+    }
+
+    try {
+        context.startActivity(Intent.createChooser(intent, "Open with"))
+    } catch (e: ActivityNotFoundException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)